src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -294,11 +294,10 @@ public TagHelperContent EndTagHelperWritingScope()
 
                 foreach (var attribute in tagHelperOutput.Attributes)
                 {
-                    var value = HtmlEncoder.HtmlEncode(attribute.Value);
                     writer.Write(' ');
                     writer.Write(attribute.Key);
                     writer.Write("=\"");
-                    writer.Write(value);
+                    WriteTo(writer, HtmlEncoder, attribute.Value, escapeQuotes: true);
```

Comment:
- `escapeQuotes` is currently only relevant if inside of an attribute value. The `WriteTo` method having an extra `escapeQuotes` parameter seems like the wrong place.  Should create a new `HtmlString` (if its an `HtmlString`) with the modified `attribute.Value` with escaped quotes here, then pass through to `WriteTo`. That way we aren't touching a piece of code that everything else calls into.
- Why pass through the `HtmlEncoder` when it's always just using the pages anyways? Suggest not passing through and just using the pages `HtmlEncoder`.